### ISSUE TITLE
Chrome 81.0.4044.113 breaks refresh of unselected items

### ISSUE THREAD

**Issue by IgorWolbers** [istvan-ujjmeszaros/bootstrap-duallistbox#160]
# Bug Report 
Using the latest version of Chrome (`81.0.4044.113`) breaks this module. See also the attached file [Dual Listbox - Chrome Bug][1], it is a gif of the reproduction using the [virtuosoft.eu][2] site.

# Reproduce
* Check that you are running the latest version of chrome `81.0.4044.113`
* Navigate to the demo page: https://www.virtuosoft.eu/code/bootstrap-duallistbox/
* In the first demo list box select one or more items by moving them from the left to the right
* Unselect 1 item by moving it from the right back to the left

## Expected result
The unselected item is now back in the list box on the left along with other unselected items

## Observed Result
- All items disappear in the unselected listbox
- The unselected item list will show content if you click in it

---
# Updates
- I have [submitted bug report on the Chromium site, #1072475][3]. Currently it has been marked as priority 1.
- The issue can be reproduced when calling `appendTo(select)` on the `select` element which occurs in `sortOptions` call.

[1]:https://user-images.githubusercontent.com/6107962/79766344-daf04800-82f5-11ea-8ea6-08187094c27c.gif
[2]:https://www.virtuosoft.eu/code/bootstrap-duallistbox/
[3]:https://bugs.chromium.org/p/chromium/issues/detail?id=1072475

**Comment by isHanz89** [comment]
hi, any hotfix for this issue?

**Comment by IgorWolbers** [comment]
You could change the `sortOptions` function as follows which fixes the issue on Chrome. As it is a bug in Chrome it might be easier to wait until they fix it, it is currently in development. See also the linked bug report in the original post.

The method `sortOptions` can be found in `/src/jquery.bootstrap-duallistbox.js` starting at line 200. Change it as follows. If the maintainer of this library would like a pull request with the change I will gladly submit one.

```
function sortOptions(select) {
  var options = select.find('option').clone(true, true).sort(function(a, b) {
    return ($(a).data('original-index') > $(b).data('original-index')) ? 1 : -1;
  });
  select.empty();
  select.append(options);
}
```

---
*Original code*
```
function sortOptions(select) {
  select.find('option').sort(function(a, b) {
    return ($(a).data('original-index') > $(b).data('original-index')) ? 1 : -1;
  }).appendTo(select);
}
```

**Comment by isHanz89** [comment]
Noted, thank you.

**Comment by IgorWolbers** [comment]
> Thanks for all the information @IgorWolbers !

Of course, and thank you for the great library! Let me know if there is anything else I can do to help with this issue.

**Comment by paulwesson** [comment]
@IgorWolbers  thank you for that fix, but it appears to have caused a new problem.

yes, when you remove an item from the list on the right, the list on the left no longer disappears, but the issue now is:

If you move an item from left to right, all items in the right hand list (other than the item you just moved) are auto selected.

Have you noticed this problem?

**Comment by IgorWolbers** [comment]
@paulwesson - no, I have not experienced that. I guess you could either wait for the fix by the Chromium team or you can create a self contained example where it does not work and post that. It would be helpful if you had a link to a stackblitz that illustrated the issue.

**Comment by balazsbonis** [comment]
@IgorWolbers @paulwesson I can confirm that issue of the auto selection, and there's a workaround of sorts, although I find it questionable in terms of quality...

Change the `sortOptions` function as follows:
```javascript
function sortOptions(select, preserveSelectionOnMove) {
        var options = select.find('option').clone(true, true).sort(function (a, b) {
            return ($(a).data('original-index') > $(b).data('original-index')) ? 1 : -1;
        });
        if (preserveSelectionOnMove !== 'all' && preserveSelectionOnMove !== 'moved') {
            options.each(function (index, item) {
                item.selected = false;
            });
        }
        select.empty();
        select.append(options);
    }
```

and then in the bottom of the `move` function, pass in this parameter as follows:
```javascript
if(dualListbox.settings.sortByInputOrder){
        sortOptionsByInputOrder(dualListbox.elements.select2);
    } else {
        sortOptions(dualListbox.elements.select2, dualListbox.settings.preserveSelectionOnMove);
    }
```

This will get rid of the auto-selection.

**Comment by istvan-ujjmeszaros** [comment]
Okay, guys, it seemed to affect a lot of users so I have released a new version with a workaround that shouldn't have any side effect (except a possible performance hit but I didn't test it - please let me know if it seems to have any negative effect).

**Comment by IgorWolbers** [comment]
Just a quick update. The Chrome team has pushed the latest release (`81.0.4044.138` at the time of writing this) which contains the fix for [bug #1072475](https://bugs.chromium.org/p/chromium/issues/detail?id=1072475). The issue has been resolved. 

I guess you can roll back your change @istvan-ujjmeszaros. Also thank you for all your work and contributions.

**Comment by istvan-ujjmeszaros** [comment]
Thanks for letting me know. I think the workaround should be there for a while, I don't have an idea on how fast Chrome will be updated for most of the users.
